### PR TITLE
Add AppVeyor CI for documentation generation

### PR DESCRIPTION
**Pull request by formulahendry**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@formulahendry__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (junhan). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by formulahendry** [comment]
Ping @BurtBiel , could you please take a look? :smiley:

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+install:
+  - python -m pip install -r requirements.txt
+  - python -m pip install -e . # Install the CLI as a package
+  - python -m pip install sphinx
+  - python scripts/command_modules/install.py # Install the command modules as packages
+
+build_script:
+  - cd doc\sphinx 
+  - make xml
+  - mkdir %TEMP%\azure-cli-xml2yml
```
Comment (BurtBiel) [inline appveyor.yml]: will the script fail if this directory already exists from a previous run?

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  - git config --global core.autocrlf false
+
+clone_depth: 5
+
+environment:
+  access_token:
+    secure: VMFbecLLHzDq/09YDPbcM0VDDSwwgY57vr5GXK6cZZ4Ti/Xs5RZoylzV8MMr1350
+
+install:
+  - python -m pip install -r requirements.txt
```
Comment (BurtBiel) [inline appveyor.yml]: if the environment doesn't get reset each time, you may want to run this install in a virtual environment.  That will cover the case when a module is removed

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+artifacts:
+  - path: doc\sphinx\_build
+
+on_success:
+  - git clone --depth 5 -q --branch=%TARGET_BRANCH% %GIT_CONTENT_REPO_URL% %TEMP%\azure-cli-content
+  - cd %TEMP%\azure-cli-content
+  - if exist %TEMP%\azure-cli-content\%YML_OUTPUT_FOLDER% rmdir /s /q %TEMP%\azure-cli-content\%YML_OUTPUT_FOLDER%
+  - SETLOCAL EnableDelayedExpansion
+  - robocopy %TEMP%\azure-cli-xml2yml\yml-output %TEMP%\azure-cli-content\%YML_OUTPUT_FOLDER% /e & IF !ERRORLEVEL! LEQ 1 exit 0
```
Comment (BurtBiel) [inline appveyor.yml]: do you mean to exit with success if robocopy returns 1, 0, -1, etc.?

File: appveyor.yml (hunk 2 above)
Comment (formulahendry) [inline appveyor.yml]: AppVeyor will use a clean new build machine for every CI, so it is not a problem.

File: appveyor.yml (hunk 1 above)
Comment (formulahendry) [inline appveyor.yml]: Same here. AppVeyor will use a clean new build machine for every CI.

File: appveyor.yml (hunk 3 above)
Comment (formulahendry) [inline appveyor.yml]: Robocopy exit code has [different meannings](http://ss64.com/nt/robocopy-exit.html). I have a new commit with minor change: we want to mark robocopy as success only if the exit code is 1 (One or more files were copied successfully).

**Comment by BurtBiel** [comment]
:shipit:

**Pull request merged**